jailbreak.c

```diff
@@ -76,27 +76,47 @@ int jbc_get_cred(struct jbc_cred* ans)
 int jbc_set_cred(const struct jbc_cred* ans)
 {
     uintptr_t td = jbc_krw_get_td();
-    uintptr_t proc = jbc_krw_read64(td+8, KERNEL_HEAP);
-    uintptr_t ucred = jbc_krw_read64(proc+0x40, KERNEL_HEAP);
-    uintptr_t fd = jbc_krw_read64(proc+0x48, KERNEL_HEAP);
-    if(ppcopyin(&ans->uid, 1+&ans->svuid, ucred+4)
-    || ppcopyin(&ans->rgid, 1+&ans->svgid, ucred+20)
-    || ppcopyin(&ans->prison, 1+&ans->prison, ucred+0x30)
-    || ppcopyin(&ans->cdir, 1+&ans->jdir, fd+0x10))
+    uintptr_t proc = jbc_krw_read64(td + 8, KERNEL_HEAP);
+    uintptr_t ucred = jbc_krw_read64(proc + 0x40, KERNEL_HEAP);
+    uintptr_t fd = jbc_krw_read64(proc + 0x48, KERNEL_HEAP);
+
+
+    if(ans->sceProcCap != NULL)
```

Comment:
Why check? The user can (not) obtain correct values for these fields by calling jbc_get_cred.